packages/core/content-type-builder/server/src/services/builder.ts

```diff
@@ -1,44 +1,95 @@
+
+  // ID fields
+  'id',
+  'document_id',
+
+  // Creator fields
+  'created_at',
+  'updated_at',
+  'published_at',
+  'created_by_id',
+  'updated_by_id',
+  // does not actually conflict because the fields are called *_by_id but we'll leave it to avoid confusion
+  'created_by',
+  'updated_by',
+
+  // Used for Strapi functionality
+  'entry_id',
+  'status',
+  'localizations',
+  'meta',
+  'locale',
+  '__component',
+  '__contentType',
+
+  // We support ending with * to denote prefixes
+  'strapi*',
+  '_strapi*',
+  '__strapi*',
+];
+
+// use snake_case
+export const reservedModels = [
+  'boolean',
+  'date',
+  'date_time',
+  'time',
+  'upload',
+  'document',
+  'then', // no longer an issue but still restricting for being a javascript keyword
+
+  // We support ending with * to denote prefixes
+  'strapi*',
+  '_strapi*',
+  '__strapi*',
+];
+
 export const getReservedNames = () => {
   return {
-    // use kebab case everywhere since singularName and pluralName are validated that way
-    models: [
-      'boolean',
-      'date',
-      'date-time',
-      'time',
-      'upload',
-      'document',
-      'then', // no longer an issue but still restricting for being a javascript keyword
-    ],
-    // attributes are compared with snake_case(name), so only snake_case is needed here and camelCase + UPPER_CASE matches will still be caught
-    attributes: [
-      // TODO: these need to come from a centralized place so we don't break things accidentally in the future and can share them outside the CTB, for example on Strapi bootstrap prior to schema db sync
-
-      // ID fields
-      'id',
-      'document_id',
-
-      // Creator fields
-      'created_at',
-      'updated_at',
-      'published_at',
-      'created_by_id',
-      'updated_by_id',
-      // does not actually conflict because the fields are called *_by_id but we'll leave it to avoid confusion
-      'created_by',
-      'updated_by',
-
-      // Used for Strapi functionality
-      'entry_id',
-      'status',
-      'localizations',
-      'meta',
-      'locale',
-      // TODO: remove these in favor of restricting the strapi_ prefix
-      'strapi',
-      'strapi_stage',
-      'strapi_assignee',
-    ],
+    models: reservedModels,
+    attributes: reservedAttributes,
   };
-  // strapi.db.getReservedNames();
+};
+
+// compare snake case to check the actual column names that will be used in the database
+export const isReservedModelName = (name: string) => {
+  const snakeCaseName = snakeCase(name);
+  if (reservedModels.includes(snakeCaseName)) {
+    return true;
+  }
+
+  if (
+    reservedModels
+      .filter((key) => key.endsWith('*'))
+      .map((key) => key.slice(0, -1))
+      .some((prefix) => snakeCaseName.startsWith(prefix))
+  ) {
+    return true;
```

Comment:
```suggestion
  const wildcardPatterns = reservedModels
    .filter((key) => key.endsWith('*'))
    .map((key) => `^${key.slice(0, -1)}`)
    .join('|');

  const regex = new RegExp(wildcardPatterns);

  if (regex.test(snakeCaseName)) {
    return true;
  }
```

Would it be better to make this test using regex?

Reply:
If we reused it or did anything else with it maybe, but I feel like in this case it's easier to understand and update the simple version. But I have no problem changing it if anyone else prefers it as a regex :)